### ISSUE TITLE
Dropdown Clear icon not working

### ISSUE THREAD

**Issue by chrisjfunk** [primefaces/primeng#14772]
### Describe the bug

enable clear button on any p-dropdown.    clicking the clear button simply drops the dropdown down again.   almost as if the z-index is off and the arrow takes precedence over the clear icon

### Environment

7.3.3 (I believe its also happening in 7.5.0)

### Angular version

17.1.2

### PrimeNG version

17.3.3

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

20.x

**Comment by rosenthalj** [comment]
This issue is the result of the PrimeNG's dropdown "Clear X Button" having a very small clickable region.

![Cursor_and_Angular_Dropdown_Component](https://github.com/primefaces/primeng/assets/45439491/dcc67062-9a4e-4e01-8939-b21bdd7cdfbf)

In the video listed below,  I used the browser's development tools to change the background color of PrimeNG's "Clear X Button"

https://github.com/primefaces/primeng/assets/45439491/77b61d2e-e203-4f5a-8d27-44d4b0f179fe

**Comment by wilson-webdev** [comment]
@chrisjfunk you're right about the `z-index`, I adjusted the `z-index` of `.p-dropdown-clear-icon` and the clear icon now works

**Comment by mehmetcetin01140** [comment]
Hi,

So sorry for the delayed response! Improvements have been made to many components recently, both in terms of performance and enhancement. Therefore, this improvement may have been developed in another issue ticket without realizing it. You can check this in the documentation. If there is no improvement on this, can you open a new issue so we can include it in our roadmap?

Thanks a lot for your understanding!
Best Regards,

**Comment by erkamyaman** [comment]
:host ::ng-deep .p-dropdown-clear-icon {
  pointer-events: auto;
}

this solved it for me
